File: javapatterns/src/main/java/org/iut/nantes/observer/meteo/AffichageStats.java
package org.iut.nantes.observer.meteo;

import java.util.ArrayList;

public class AffichageStats implements Observer, Affichage{
    ArrayList<Integer> temps;
    ArrayList<Integer> humidities;
    ArrayList<Integer> pressions;

    public AffichageStats(Sujet sujet) {
        sujet.addObserver(this);
        temps = new ArrayList<>();
        humidities = new ArrayList<>();
        pressions = new ArrayList<>();
    }

    @Override
    public void afficher() {
        System.out.println("Temperature: " + temps.stream().mapToInt(value -> value).average().getAsDouble());
        System.out.println("Humidity: " + humidities.stream().mapToInt(value -> value).average().getAsDouble());
        System.out.println("Pression: " + pressions.stream().mapToInt(value -> value).average().getAsDouble());
    }

    @Override
    public void updateData(int temp, int humidity, int pression) {
        temps.add(temp);
        humidities.add(humidity);
        pressions.add(pression);
        afficher();
    }
}

File: src/fr/tblf/patterns/observer/meteo/AffichageStats.kt
package fr.tblf.patterns.observer.meteo

class AffichageStats(data: DonneesMeteo) : Observer, Affichage(data) {

    val temps = ArrayList<Float>()
    val humidites = ArrayList<Float>()
    val pressions = ArrayList<Float>()

    init {
        data.enregistrerObserver(this)
    }

    override fun update(temp: Float, humidite: Float, pression: Float) {
        temps.add(temp)
        humidites.add(humidite)
        pressions.add(pression)
        afficher()
    }

    override fun afficher() {
        println("Temperature moyenne: ${temps.average()} °C \nhumiditée moyenne: ${humidites.average()} %  \npression moyenne: ${pressions.average()} hPa")
    }
}